nodeadm/internal/configprovider/userdata.go

```diff
@@ -29,7 +31,12 @@ type userDataConfigProvider struct {
 
 func NewUserDataConfigProvider() ConfigProvider {
 	return &userDataConfigProvider{
-		imdsClient: imds.New(imds.Options{}),
+		imdsClient: imds.New(imds.Options{
+			Retryer: retry.NewStandard(func(so *retry.StandardOptions) {
+				so.MaxAttempts = 15
+				so.MaxBackoff = 1 * time.Second
```

Comment:
by default `imds.New` uses the standard retryer but sets `MaxBackoff` as `1` and does not change the `MaxAttempts` from `3`, so we're just mimicking this and bumping the attempts